platform/android/java/lib/src/org/godotengine/godot/input/GodotEditText.java

```diff
@@ -155,13 +155,22 @@ public void setView(final GodotRenderView view) {
 	// ===========================================================
 	@Override
 	public boolean onKeyDown(final int keyCode, final KeyEvent keyEvent) {
-		super.onKeyDown(keyCode, keyEvent);
+		// pass event to godot in special cases
+		if (keyCode == KeyEvent.KEYCODE_DPAD_UP || keyCode == KeyEvent.KEYCODE_DPAD_DOWN ||
+				keyCode == KeyEvent.KEYCODE_DPAD_LEFT || keyCode == KeyEvent.KEYCODE_DPAD_RIGHT ||
+				// shift key should used to uppercase by EditText
+				(KeyEvent.isModifierKey(keyCode) && keyCode != KeyEvent.KEYCODE_SHIFT_LEFT && keyCode != KeyEvent.KEYCODE_SHIFT_RIGHT) ||
+				keyEvent.isAltPressed() || keyEvent.isCtrlPressed() || keyEvent.isSymPressed() ||
+				keyEvent.isFunctionPressed() || keyEvent.isMetaPressed()) {
+			mRenderView.getView().onKeyDown(keyCode, keyEvent);
+		} else {
+			super.onKeyDown(keyCode, keyEvent);
```

Comment:
There are a few things I don't understand in this block:
- What does `mRenderView.getView().onKeyDown(keyCode, keyEvent);` do compared to `super.onKeyDown(keyCode, keyEvent);`? It's not clear to me how it passes events to godot in special cases this way.
- When `KeyEvent.isModifierKey(keyCode)` is true and it's not shift, isn't it already handling all the other modifier cases?
- Why is shift the only modifier not handled by this special case?

Reply:
EditText is an overlay on top on the activity (as I understand) so any event that happen while it's up doesn't get sent to Godot, so in case EditText doesn't handle it as we want it to, I send it back to Godot (which is what renderView does).

Shift is handled in EditText by uppercasing the text so we'll let EditText to handle that

all other modifier like control godot need to handle so shortcuts hanled correctly

Reply:
I'm not 100% sure about shift special handling as shift arrow key (select text) doesn't work because of that

Reply:
Thanks for the details. Here are a few things to improve this part of the code:
1. You can make a separate helper function `is_special_key` so you can re-use the same logic in both `onKeyDown` and `onKeyUp` without duplicating the code.
2. You don't need to test specifically for Alt, Ctrl, etc. here because `isModiferKey` already handles these cases:
https://developer.android.com/reference/android/view/KeyEvent#isModifierKey(int)
3. You should probably call `onKeyDown` and `onKeyUp` on the input handler directly instead of the view if the only purpose is to pass them to Godot. It will avoid unwanted side effects.
4. In order to handle shift correctly for selection purpose: what about passing shift modifier to Godot as well (it will handle the uppercase too), but also pass all modifiers only in the case of a physical keyboard? It would be cleaner and safer this way. Is there an easy way to know if we're using a virtual or physical keyboard?

Reply:
about 2. I need to test if ctrl is held in case of ctrl + key, so the second key would be sent to godot instead of handled in EditText
and yeah, I'll probebly do that and handle any speicial key in godot, it will make it easier

Reply:
also Im adding tab to here (the actual thing I need to test is is that a char)

Reply:
also Im adding tab to here (the actual thing I need to test is is that a char)

Reply:
as for 3 I think we want the "side effects" here, we just want that if EditText doesn't need to handle that, than it should be handled like we are not in that state

Reply:
Just tested and it looks like godot doesn't handle uppercase the text

Reply:
For 2. Ok, I've misread and it makes sense actually. It would help readability though to add a comment to explain that you need to pass "modifier + key" events to Godot as well.

For 3. I find this a little weird, but I'll let @m4gr3d decide if it's ok.

For 4. Then you might have to make it so when modifier keys are pressed (or possibly just shift) you pass it to Godot and to the text edit. I also still think it would be better to do this special key handling only for physical keyboards.

Reply:
4. https://developer.android.com/training/keyboard-input/commands 
> Note: When handling keyboard events with the KeyEvent class and related APIs, you should expect that such keyboard events come only from a hardware keyboard. You should never rely on receiving key events for any key on a soft input method (an on-screen keyboard).

Reply:
hmm.. I thinking about 4...  if it just the modifiers I don't think it should be a problem, Ill test that

Reply:
Ok, changed that it sends all modifier, but sends key only if non shift modifier is pressed, it does seem to work

Reply:
BTW, before @m4gr3d comment on this, I tried sending false in those cases instead of passing the event and it didn't work

Reply:
> BTW, before @m4gr3d comment on this, I tried sending false in those cases instead of passing the event and it didn't work

Passing `false` doesn't work because the `GodotInputHandler` is attached to the `RenderView` instances which are at the [same level](https://github.com/godotengine/godot/blob/9e34ba48556b71f1f672b5be47d12b0e3f679b86/platform/android/java/lib/src/org/godotengine/godot/Godot.java#L226) as the `GodotEditText` view; the event goes [to the parent view](https://github.com/godotengine/godot/blob/9e34ba48556b71f1f672b5be47d12b0e3f679b86/platform/android/java/lib/src/org/godotengine/godot/Godot.java#L219) instead.